NBitcoin/Crypto/Hashes.cs

```diff
@@ -801,17 +801,29 @@ public static byte[] HMACSHA512(byte[] key, byte[] data)
 		{
 			var mac = new NBitcoin.BouncyCastle.Crypto.Macs.HMac(new Sha512Digest());
 			mac.Init(new KeyParameter(key));
-			mac.Update(data);
+			mac.BlockUpdate(data, 0, data.Length);
 			byte[] result = new byte[mac.GetMacSize()];
 			mac.DoFinal(result, 0);
 			return result;
 		}
+#if HAS_SPAN
+		public static bool HMACSHA512(byte[] key, ReadOnlySpan<byte> data, Span<byte> output, out int outputLength)

```

Comment:
the code does not seems correct

Reply:
fixed it on https://github.com/MetacoSA/NBitcoin/commit/9326fdb18f1ecdb2388f1136440ab17315de9483